File: xmlparser.rb
require 'sitemap-parser'
require 'csv'
sitemap = SitemapParser.new "https://newdistrict.ca/sitemap.xml"
sitemap.urls # => Array of Nokigiri XML::Node objects
sitemap.to_a # => Array of url strings

link_array = sitemap.to_a

puts link_array

link_array.each do |i|
	CSV.open("file.csv", "a+") do |csv|
	  csv << [i]
  end
end

File: xmlparser.php
<?php
$urls = array();
$sitename = "New District";

$DomDocument = new DOMDocument();
$DomDocument->preserveWhiteSpace = false;
$DomDocument->load('https://newdistrict.ca/sitemap.xml');
$DomNodeList = $DomDocument->getElementsByTagName('loc');

foreach($DomNodeList as $url) {
    $urls[] = $url->nodeValue;
}

$file = fopen("$sitename.csv","w");

foreach ($urls as $line)
  {
  fputcsv($file,explode(',',$line));
  echo $line . "\n";

  }

fclose($file); ?>


?>
